collated/main/A0111277M.md

```diff
@@ -0,0 +1,594 @@
+    
+    private static final String START_DATE = "startDate" ;
+    private static final String END_DATE = "endDate" ;
+    private List<ReadOnlyTask> tasksAdded = Lists.newLinkedList();
+    private List<ReadOnlyTask> tasksDeleted = Lists.newLinkedList();
+    private final int targetIndex;
+    private final String name;
+    private final String description;
+    private boolean doneStatus;
+    private final Set<String> tagNameSet;
+    private final Map<String, LocalDateTime> dateMap ;
+    
+    private boolean hasChanged = false ;
+    
+    public EditCommand(int targetIndex, String name, String description, Set<String> tags, LocalDateTime startDate, LocalDateTime endDate) throws IllegalValueException {
+        this.targetIndex = targetIndex;
+        this.name = name;
+        this.description = description;
+        this.tagNameSet = tags ;
+        this.dateMap = Maps.newHashMap() ;
+        
+        if (startDate != null) {
+            dateMap.put(START_DATE, startDate) ;
+        }
+        
+        if (endDate != null) {
+            dateMap.put(END_DATE, endDate) ;
+        }
+
+    }
+    
+    public EditCommand(int targetIndex, String name, String description, Set<String> tags) throws IllegalValueException {
+        this (targetIndex, name, description, tags, null, null) ;
+    }
+    
+    public EditCommand(int targetIndex, String name, String description, Set<String> tags, LocalDateTime endDate) throws IllegalValueException {
+        this (targetIndex, name, description, tags, null, endDate) ;
+    }
+    
+    @Override
+    public CommandResult execute() {
+        String newName;
+        String newDescription;
+        UniqueTagList newTagSet;
+        UnmodifiableObservableList<ReadOnlyTask> lastShownList = model.getFilteredTaskList();
+        Task newTask;
+        
+        /* determine target task to delete based on lastShownList */
+        if (lastShownList.size() < targetIndex || targetIndex < 1) {
+            indicateAttemptToExecuteIncorrectCommand();
+            return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+        
+        ReadOnlyTask taskToEdit = lastShownList.get(targetIndex - 1);  
+    
+        newName = checkUpdate(taskToEdit.getName(), name);
+        
+        determineDateTimeOfNewTask(taskToEdit);
+        
+        if (taskToEdit instanceof Block) {
+        	
+        	if (dateMap.get(START_DATE) == null || dateMap.get(END_DATE) == null) {
+        		return new CommandResult(MESSAGE_BLOCK_CANNOT_REMOVE_DATE);
+        	}
+        	
+        	newTask = new Block(taskToEdit.getTaskId(), newName, dateMap.get(START_DATE), dateMap.get(END_DATE));
+        	
+        } else {
+        	
+        	if (dateMap.get(START_DATE) != null && dateMap.get(END_DATE) == null) {
+        		return new CommandResult(MESSAGE_CANNOT_HAVE_ST_ONLY);
+        	}
+
+	        
```

Comment:
The line gaps are too much. do not have line spaces unless the code snippet is relating to a different function, or sub purpose